tests/test_all_fields.py

```diff
@@ -548,8 +585,14 @@ def all_fields_test(self, streams_to_test):
                                         f"AssertionError({failure_1})")
 
                                     nested_key = KNOWN_NESTED_MISSING_FIELDS.get(stream, {})
-                                    if self.find_nested_key(nested_key, expected_field_value, field):
-                                        continue
+                                    # Check whether expected_field_value is list or not.
+                                    # If expected_field_value is list then loop through each item of list
+                                    if type(expected_field_value) == list:
+                                        if self.handle_list_data(expected_field_value, field, nested_key):
+                                            continue
+                                    else:
+                                        if self.find_nested_key(nested_key, expected_field_value, field):
+                                            continue
```

Comment:
I do not understand what `handle_list_data` and `find_nested_key` are trying to accomplish. All I see them do is skip the checks below. I think it adds complexity to the test and removes the test coverage below and the logged known failures.

Reply:
@kspeer There were 2 types of nested fields replicated from the API, 1. dict and 2. list. So, here we have handled dict keys in `find_nested_key` and ignore the failing nested fields and for list fields, we loop over the fields and ignore failing nested fields. Previously we had skipped the first-level field directly so we added the above change to ignore only failed nested fields.

Reply:
We were already doing that based on hardcoding expectations. Now if new fields start failing we will not know about it. We should purposefully document and skip discrepant fields if there subfields differ using the already implemented hardcoding approach. We should do this programmatically.

Reply:
Got it @kspeer825. Update the all fields test case.